src/extensions/default/BrambleUrlCodeHints/main.js

```diff
@@ -155,13 +155,48 @@ define(function (require, exports, module) {
                         return;
                     }
 
+                    function getFilesRecursively(targetDir, callback) {
+                        var directory = FileSystem.getDirectoryForPath(targetDir);
+                        var unfiltered = [];
+                        directory.getContents(function (err, contents) {
```

Comment:
You're mixing synchronous (line 153) and asynchronous loops here, and it will lead to timing issues.  Async looping is done using the`src/util/Async` module.  Also, you're duplicating the code that loops through a directory's contents: why not just call it recursively?